src/Hpack/Config.hs

```diff
@@ -1265,9 +1275,11 @@ toSection = go
       , sectionDependencies = fromMaybe mempty commonOptionsDependencies
       , sectionPkgConfigDependencies = fromMaybeList commonOptionsPkgConfigDependencies
       , sectionConditionals = map toConditional (fromMaybeList commonOptionsWhen)
-      , sectionBuildTools = fromMaybe mempty commonOptionsBuildTools
+      , sectionBuildTools = maybe mempty toBuildToolMap commonOptionsBuildTools
```

Comment:
`maybe mempty f` === `foldMap f`, which may or may not be an improvement

Reply:
Also, this silently has first-one-wins behaviour when the same build tool is specified multiple times. Worth documenting if nothing else.

Reply:
It should be last one wins?!

Reply:
The `Monoid` instance of `Map` uses `union`, which is left-biased. If it's going to be last-one-wins, then we'll have to use `Dual` or `reverse` the input, then.

Reply:
`BuildTools` are still a list when the concatenation happens.   So the `Monoid` instance for `[]` is used.  This is to make sure that the cases

```
gives per-section unqualified names precedence over global qualified names
```
and
```
gives per-section qualified names precedence over global unqualified names
```
are met.

These two tests also test that we indeed have last one wins behavior.